### PR TITLE
History page

### PR DESCRIPTION
**Pull request by foolhardy21**
## History Page

This PR has the history page components and APIs. The videos are added to history when they are played by logged in user.

 - Files to review:
   - [ ] src/components/History/HistorySection.js
   - [ ] src/components/History/HistoryVideoCard.js
   - [ ] src/pages/History.js
   - [ ] src/contexts/history.context.js
   - [ ] src/reducers/history.reducer.js

- Use this link for preview - https://deploy-preview-4--pensive-hodgkin-f2cf81.netlify.app/history

v-1

Thanks.

### CODE REVIEW

File: src/reducers/signup.reducer.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
+export function signupFormReducer(state, action) {
+
+    switch (action.type) {
```
Comment (MayankKumar10) [inline src/reducers/signup.reducer.js]: you can destructure action  further into {type, payload}

File: public/manifest.json [hunk 2]
```diff
@@ -1,23 +1,6 @@
...
-      "type": "image/png",
-      "sizes": "192x192"
-    },
-    {
-      "src": "logo512.png",
-      "type": "image/png",
-      "sizes": "512x512"
-    }
-  ],
   "start_url": ".",
```
Comment (MayankKumar10) [inline public/manifest.json]: is it required

File: src/components/Videos/VideoCard.js [hunk 3]
```diff
@@ -12,7 +12,7 @@ const VideoCard = ({ video: {
     return (
         <Card id='container-video' classes='pd-xs pos-relative'>
 
-            <Button classes={`btn-solid ${getSolidBtnBgColor(theme)} ${getSolidBtnTextColor(theme)} txt-md txt-lcase pd-xs pos-absolute tr-1`}>watch later</Button>
+            {/* <Button classes={`btn-solid ${getSolidBtnBgColor(theme)} ${getSolidBtnTextColor(theme)} txt-md txt-lcase pd-xs pos-absolute tr-1`}>watch later</Button> */}
```
Comment (MayankKumar10) [inline src/components/Videos/VideoCard.js]: remove this comment

**Review by MayankKumar10 — APPROVED** [review]
Look into these Suggestions, everything else looks good to me

File: src/components/History/HistoryVideoCard.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    return (
+        <Card id='container-video' classes='pd-xs pos-relative'>
+
+            <video id='card-video' controls controlsList="nodownload nofullscreen">
+                <source src={url}></source>
+            </video>
+
+            <Text classes={`txt-md txt-cap txt-500 ${getTextColor(theme)} card-txtw-s mg-btm-xs`}>{title}</Text>
+
+            <Text classes={`txt-md txt-cap ${getTextColor(theme)} card-txtw-s`}>{description.slice(0, 40)}....</Text>
```
Comment (anik31) [inline src/components/History/HistoryVideoCard.js:35]: appreciation - liked that you have broken everything into components

File: src/contexts/history.context.js [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+        <HistoryContext.Provider
+            value={{
+                historyState,
+                historyDispatch,
+                getHistory,
+                removeHistory,
+                showHistoryAlert,
+                addVideoToHistory,
+                removeVideoFromHistory,
+            }}
```
Comment (anik31) [inline src/contexts/history.context.js:95]: appreciation - liked that you have used a separate context and defined the API call functions inside it

File: src/pages/History.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+            }
+        })()
+    }, [])
+
+    async function handleRemoveHistory() {
+        const removeHistoryResponse = await removeHistory()
+        if (removeHistoryResponse === 404 || removeHistoryResponse === 500) {
+            showHistoryAlert('could not remove history', 'error')
+        } else {
+            historyDispatch({ type: 'REMOVE_HISTORY' })
```
Comment (anik31) [inline src/pages/History.js:34]: appreciation - it's good that you're doing error handling as well

**Review by anik31 — APPROVED** [review]
Nice work, code is clean & well structured

File: src/reducers/signup.reducer.js (hunk 1 above)
Comment (foolhardy21) [reply src/reducers/signup.reducer.js]: made the changes. thanks.

File: public/manifest.json (hunk 2 above)
Comment (foolhardy21) [reply public/manifest.json]: no, removed.

File: src/components/Videos/VideoCard.js (hunk 3 above)
Comment (foolhardy21) [reply src/components/Videos/VideoCard.js]: removed. thanks.

File: src/components/History/HistoryVideoCard.js (hunk 4 above)
Comment (foolhardy21) [reply src/components/History/HistoryVideoCard.js:35]: thankyou

**Pull request merged**